sympy/matrices/matrices.py

```diff
@@ -2402,21 +2435,21 @@ def dot(self, b):
                     type(b))
 
         mat = self
-        if mat.cols == b.rows:
-            if b.cols != 1:
-                mat = mat.T
-                b = b.T
-            prod = flatten((mat * b).tolist())
-            if len(prod) == 1:
-                return prod[0]
-            return prod
-        if mat.cols == b.cols:
-            return mat.dot(b.T)
-        elif mat.rows == b.rows:
-            return mat.T.dot(b)
-        else:
-            raise ShapeError("Dimensions incorrect for dot product: %s, %s" % (
-                self.shape, b.shape))
+        if (1 not in mat.shape) or (1 not in b.shape) :
+            SymPyDeprecationWarning(
+                feature="dot() no longer support dot of (m,n) and (1,n) matrices.",
+                issue=13808,
```

Comment:
Create a issue stating this behavior is deprecated and will be removed in the future releases, and reference that issue number. See https://github.com/sympy/sympy/issues?q=is%3Aopen+is%3Aissue+label%3A%22Deprecation+Removal%22 for examples.

Reply:
yup sure!